packages/sidecar/src/mcp/index.ts

```diff
@@ -34,34 +34,35 @@ export function createMcpInstance(buffer: MessageBuffer<Payload>) {
         };
       }
 
-      const errors = envelopes
+      const content = envelopes
         .map(([contentType, data]) => processEnvelope({ contentType, data }))
         .sort((a, b) => {
           const a_sent_at = a.envelope[0].sent_at as string;
           const b_sent_at = b.envelope[0].sent_at as string;
           if (a_sent_at < b_sent_at) return 1;
           if (a_sent_at > b_sent_at) return -1;
           return 0;
+        })
+        .flatMap(processed => {
```

Comment:
Let's move `.flatMap()` before sort so we don't spend time sorting filtered items? :)

Reply:
Actually, moving this later makes sorting proper so it should be part of the patch.